packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js

```diff
@@ -418,6 +430,39 @@ class EnumerationWidget extends React.Component {
 		});
 	}
 
+	onValidateAndAddHandler(event, value) {
+		if (!value.value) {
+			this.setState({
+				displayMode: DISPLAY_MODE_DEFAULT,
+			});
+			return;
+		}
+
+		if (this.callActionHandler(
+				ENUMERATION_ADD_ACTION,
+				this.parseStringValueToArray(value.value),
+				this.validateAndAddSuccessHandler.bind(this),
+				this.addFailHandler.bind(this))
+		) {
+			this.setState({
+				headerInput: this.loadingInputsActions,
+			});
+			this.input.focus();
+		} else if (!this.valueAlreadyExist(value.value)) {
+			this.setState(
+				{
+					items: this.state.items.concat([{
```

Comment:
using setState this wayn don't garanties that this.state.items is in intended state. see https://facebook.github.io/react/docs/react-component.html#setstate